flink-python/src/main/java/org/apache/flink/streaming/api/operators/python/AbstractPythonFunctionOperator.java

```diff
@@ -320,23 +322,22 @@ private void checkInvokeFinishBundleByCount() throws Exception {
 	 */
 	private void checkInvokeFinishBundleByTime() throws Exception {
 		long now = getProcessingTimeService().getCurrentProcessingTime();
-		if (now - lastFinishBundleTime >= maxBundleTimeMills) {
+		if (now - lastFinishBundleTime >= maxBundleTimeMills && elementCount > 0) {
 			invokeFinishBundle();
 		}
 	}
 
-	private void invokeFinishBundle() throws Exception {
-		if (bundleStarted.compareAndSet(true, false)) {
-			pythonFunctionRunner.finishBundle();
-
-			emitResults();
-			elementCount = 0;
-			lastFinishBundleTime = getProcessingTimeService().getCurrentProcessingTime();
-			// callback only after current bundle was fully finalized
-			if (bundleFinishedCallback != null) {
-				bundleFinishedCallback.run();
-				bundleFinishedCallback = null;
-			}
+	protected void invokeFinishBundle() throws Exception {
```

Comment:
Originally, the flush logic is handled inside the lock `bundleStarted`. Why the lock is removed?